src/cstring.rs

```diff
@@ -0,0 +1,41 @@
+use std::{ffi::{self}, fmt};
+
+use serde::{de::{self, Visitor}, Deserializer, Serializer};
+
+
+pub type CString = ffi::CString;
+
+pub fn serialize<S>(string: &CString, serializer: S) -> Result<S::Ok, S::Error>
+where
+    S: Serializer,
+{
+    serializer.serialize_str(&string.to_string_lossy())
+}
+
+
+struct CStringVisitor;
+
+impl Visitor<'_> for CStringVisitor {
+    type Value = CString;
+
+    fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+        formatter.write_str("a string")
+    }
+
+    fn visit_str<E>(self, value: &str) -> Result<Self::Value, E>
+    where
+        E: de::Error,
+    {
+        match CString::new(value) {
+            Ok(v) => Ok(v),
+            Err(_) => Err(E::custom("a"))
+        }
+    }
```

Comment:
This falls into the UTF-8 trap from before. Would `visit_bytes` be better? Would we want to add our own null byte and/or strip existing ones?

If not, this might be better:

```suggestion
    fn visit_str<E>(self, value: &str) -> Result<Self::Value, E>
    where
        E: de::Error,
    {
        CString::new(value).map_err(|_| E::invalid_value("invalid CString", "a valid CString"))
    }
```

Reply:
Nevermind I am confused and stupid